src/vm/eventpipebuffermanager.cpp

```diff
@@ -710,21 +718,17 @@ void EventPipeBufferManager::SuspendWriteEvent()
     }
     CONTRACTL_END;
 
-    CQuickArrayList<EventPipeThread*> threadList;
+    CQuickArrayList<EventPipeThread *> threadList;
     {
         SpinLockHolder _slh(&m_lock);
-        m_writeEventSuspending.Store(TRUE);
-        // From this point until m_writeEventSuspending is reset to FALSE it is impossible
+        // From this point onward, it is impossible
```

Comment:
This comment is like a proof of correctness and setting the flag is a required initial condition for the proof to be accurate. You could refer to the flag that has now been moved to EventPipeSession, but without a promise that the flag is TRUE the conclusions don't hold. A thread could call into AllocateBufferForThread(), take the m_lock, observe m_writeEventSuspending is FALSE, and then it would allocate a buffer.